lib/dom.js

```diff
@@ -1139,10 +1139,11 @@ function needNamespaceDefine(node, isHTML, visibleNamespaces) {
  * Well-formed constraint: No < in Attribute Values
  * The replacement text of any entity referred to directly or indirectly in an attribute value must not contain a <.
  * @see https://www.w3.org/TR/xml/#CleanAttrVals
- * @see https://www.w3.org/TR/xml/#NT-AttValue
+  * @see https://www.w3.org/TR/xml/#NT-AttValue
+  * @see https://www.w3.org/TR/xml/#AVNormalize
  */
 function addSerializedAttribute(buf, qualifiedName, value) {
-	buf.push(' ', qualifiedName, '="', value.replace(/[<&"]/g,_xmlEncoder), '"')
+	buf.push(' ', qualifiedName, '="', value.replace(/[<&"\t\n\r]/g,_xmlEncoder), '"')
```

Comment:
nit: add a missing space here

```suggestion
	buf.push(' ', qualifiedName, '="', value.replace(/[<&"\t\n\r]/g, _xmlEncoder), '"')
```